hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/insights/om/om.tsx

```diff
@@ -502,13 +512,14 @@ export class Om extends React.Component<Record<string, object>, IOmdbInsightsSta
         })
       }
       else {
-        if (this.state.prevKeyMismatch === 0 ){
+        if (this.state.prevKeyMismatch === 0 || this.state.pageDisplayCount === 1){
           this.setState({
             prevClickable: false
           })
         }
-        if (mismatchPrevKeyList.includes(mismatchContainersResponse.data.lastKey) === false) {
-          mismatchPrevKeyList.push(mismatchContainersResponse.data.lastKey);
+        //Need to avoid rewrite in Map so avoiding duplication and wrong value for
+        if (!prevKeyListMap.has(this.state.pageDisplayCount)) {
```

Comment:
I don't understand this, a map can't have duplicate keys.

Reply:
For Each Page we are assigning unique values Fox example page 1 =Last key 10, page 2, Last key=20 so we can navigate properly after clicking Previous page and Next page or limit change.

Reply:
But what do you mean by this `Need to avoid rewrite in Map so avoiding duplication and wrong value for`?

Reply:
When we change Page SIze each time we get different last keys depending upon page size so need to avoid rewrire of keys so used logic.

Reply:
As far as I understand in this map the keys are the number of the page (eg. 1, 2, 3, ...) and the values are the previous page's last key (10, 20, ...), in case of the first page it will be null. This will store this for one tab (eg. container mismatch) and once we navigate to a new tab it will be cleared up.
Into this map we are only putting key-value pairs, if previously the key wasn't there. 
So default the limit is set to 10 and if we navigate through all the pages we will fill up this map with e.g. these values:
```
1: empty
2: 10
3: 20
4: 30
5: 30
```
After this if we set the limit to 20, we won't change anything in this map. So on the 3rd page will show the elements 21-40 and for the 4th page we will show 31-50, based on this logic. But that is incorrect, as we are showing the elements from 31-40 on both pages. Could you double check this?